beavy_modules/url_extractor/lib/fetching_test.py

```diff
@@ -40,31 +41,132 @@ def __eq__(self, other):
                 return False
         return True
 
+class URL:
+
+    def scheme(self, inp):
+        self._scheme = inp
+        return self
+
+    def netloc(self, inp):
+        self._netloc = inp
+        return self
+
+    def path(self, inp):
+        self._path = inp
+        return self
+
+    def hostname(self, inp):
+        self._hostname = inp
+        return self
+
+    def port(self, inp):
+        self._port = inp
+        return self
+
+    def username(self, inp):
+        self._username = inp
+        return self
+
+    def password(self, inp):
+        self._password = inp
+        return self
+
+    def params(self, inp):
+        self._params = inp
+        return self
+
+    def scheme_check(self, inp):
+        self._scheme_check = inp
+        return self
+
+    def netloc_check(self, inp):
+        self._netloc_check = inp
+        return self
+
+    def path_check(self, inp):
+        self._path_check = inp
+        return self
+
+    def hostname_check(self, inp):
+        self._hostname_check = inp
+        return self
+
+    def port_check(self, inp):
+        self._port_check = inp
+        return self
+
+    def username_check(self, inp):
+        self._username_check = inp
+        return self
+
+    def password_check(self, inp):
+        self._password_check = inp
+        return self
+
+    def params_check(self, inp):
+        self._params_check = inp
+        return self
+
+    def __eq__(self, other):
```

Comment:
Just a joke... I didn't realize the use of self/other is key, but it makes sense. I like it!